source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,196 @@
+Install Git
+```bash
+# sudo apt-get install git
+```
+Fork and clone spksrc, this will make a directory named "spksrc".
+```bash
+$ git clone https://github.com/SynoCommunity/spksrc.git
 ```
 
+Edit the following 2 files.
 
-Check the path to python3 (assumed to be /volume1/@appstore/py3k/usr/local/bin)
+These edits will make cross compiled Python module packages, which Home Assistant needs to install. These edits also enable use of the "[Cloud](/components/cloud/)" and [Homekit](/components/homekit/)" components.
+Most components will setup just fine, but certain components need a python package that compiles on installation, which will fail as Synology did not provide a compiler, causing said need for cross compilation.
 
-```bash
-# cd /volume1/@appstore/py3k/usr/local/bin
+Edit "*~/spksrc/spk/python3/src/requirements.txt*" and add the following text to the end of the file:
 ```
+## It may happen you want to add a component to Home Assistant, but it fails to install a Python package.
+## In the logs you see a error "Unable to install package *Module_Name*", if you install that
+# package with pip, some of its required packages fail with error "Failed building wheel for *Module_Name*"
+## Add these Python modules/packages that fail with that "building wheel" error in to the list below.
 
-Install PIP (Python's package management system)
+# Example format of module:
+#pythonmodule==version
 
-```bash
-# ./python3 -m ensurepip
-```
+## Cross compilation requirements for installing Home Assistant (Tested to work on 84.3).
+cffi==1.11.5
+bcrypt==3.1.4
+cryptography==2.3.1
 
-Use PIP to install Homeassistant package 0.64.3
+## Cross compilation requirement for installing "Warrent" module (Needed by "Cloud" Component)
+pycryptodome==3.7.2
 
-```bash
-# ./python3 -m pip install homeassistant==0.64.3
+## Cross compilation requirement for installing "HAP_python" module (Needed by "Homekit" Component)
+curve25519-donna==1.3
+ed25519==1.4
 ```
 
-Create homeassistant config directory & switch to it
-
+Add these edits to fix OpenSSL errors when using the "[Xiaomi_Aqara](/components/Xiaomi_Aqara/)" component.  
+Edit "*~/spksrc/spk/python3/Makefile*" and add the following text above the text "**include ../../mk/spksrc.spk.mk**":
+```makefile
+## Makefile variable needed to fix "_openssl.so: undefined symbol: pthread_atfork" error when using "xiaomi_aqara" component.
+export CFLAGS=-pthread
+```
+#### {% linkable_title Compiling the Python 3 package %}
+Make the Python 3 package for your Synology model, please replace "arch-**XXXX**" by the apporiate architecture of your Synology. For a list of architectures, look at this [list of architectures](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model) accepted by spksrc. Depending on your computer, compilation may take a hour or more (Significantly less if you have a SSD and a moderately good CPU).
 ```bash
-# mkdir /volume1/homeassistant
```

Comment:
`mkdir -m 0755 /volume1/homeassistant`

Reply:
Making a directory manually was a part of the old guide.
The instruction for the current guide is to make a "homeassistant" Shared-Folder.